official/utils/accelerator/tpu.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import time
+
+import tensorflow as tf
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import embedding_ops
+from tensorflow.python.ops import math_ops
+
+
+DRY_RUN = "dry_run"
```

Comment:
I wonder if the `dry_run` name is actually necessary. I think passing in `--tpu=local` and `--use_tpu=False` has the same effect without the extra code needed to check.

Reply:
Yeah, I'm not sure that how I put this in is the cleanest. (Truth be told it was for my own debugging as much as anything else.) The difference between this and say, mnist_tpu.py is that use_tpu=False is not just a debug flag in Transformer since the TPU and non-TPU loops are the same. So I want use_tpu to be True even during a dry run. I like `--tpu=local` better than `--tpu=dry_run` though.

Reply:
Yeah, part of the reason of using `--tpu=local` is that I think you can remove the `if tpu == "dry_run": don't use cluster_resolver` stuff, since cluster resolver accepts an empty string or "local" and won't try to resolve those to a TPU name.

Reply:
Otherwise I think it looks OK after resolving karmel and k-w-w's comments.